### PR TITLE
feat: beginner for text annotation

### PR DESCRIPTION
**Pull request by bvanjoi**
https://github.com/web-infra-dev/rspress/assets/30187863/088b6968-0b4b-4c49-b12c-3a390bb0824e

### CODE REVIEW

File: packages/plugin-annotation/package.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
+{
+  "name": "@rspress/plugin-annotation",
+  "version": "0.0.6",
```
Comment (sanyuan0704) [inline packages/plugin-annotation/package.json:3]: The version should be updated to latest v0.0.10

File: packages/plugin-annotation/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import path from 'path';
+import type { RspressPlugin } from '@rspress/shared';
+
+const annotationComponent = path.resolve(__dirname, '../static/index.tsx');
+
+export function pluginAnnotation(): RspressPlugin {
+  return {
+    name: '@modern-js/doc-plugin-annotation',
```
Comment (sanyuan0704) [inline packages/plugin-annotation/src/index.ts:8]: @rspress/plugin-annotation

File: packages/plugin-annotation/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import path from 'path';
+import type { RspressPlugin } from '@rspress/shared';
+
+const annotationComponent = path.resolve(__dirname, '../static/index.tsx');
```
Comment (sanyuan0704) [inline packages/plugin-annotation/src/index.ts:4]: Should think of windows path, refer https://github.com/web-infra-dev/rspress/blob/main/packages/plugin-medium-zoom/src/index.ts

File: packages/core/src/theme-default/layout/DocLayout/index.tsx [hunk 4]
```diff
@@ -83,12 +83,14 @@ export function DocLayout(props: DocLayoutProps) {
           {isOverviewPage ? (
             <Overview />
           ) : (
-            <div className="rspress-doc">
-              <TabDataContext.Provider value={{ tabData, setTabData }}>
-                <MDXProvider components={getCustomMDXComponent()}>
-                  <Content />
-                </MDXProvider>
-              </TabDataContext.Provider>
+            <div>
```
Comment (sanyuan0704) [inline packages/core/src/theme-default/layout/DocLayout/index.tsx:86]: Why is there an extra container div?

**Comment by sanyuan0704** [comment]
The e2e test case seems failed.

File: packages/core/src/theme-default/layout/DocLayout/index.tsx (hunk 4 above)
Comment (bvanjoi) [reply packages/core/src/theme-default/layout/DocLayout/index.tsx:86]: We've removed all child elements of the `﻿.rspress-doc`, and I think the ﻿footer should not be considered part of the article. Thus, I add a `div` to keep `<DocFooter/>`

see: https://github.com/web-infra-dev/rspress/pull/34/files#diff-87b592124a448b2d70388b0cfb70aa567357fbb6140d9de24df2ed6ba5440778R19-R21

**Comment by chenjiahan** [comment]
This PR is outdated so I'm closing it

**Pull request closed without merging**